packages/hardhat-truffle5/src/index.ts

```diff
@@ -136,9 +136,10 @@ extendEnvironment((env) => {
 
         accounts = normalizeHardhatNetworkAccountsConfig(
           netConfig.accounts
-        ).map((acc) =>
-          toChecksumAddress(bufferToHex(privateToAddress(acc.privateKey)))
-        );
+        ).map((acc) => {
+          const buffer = Buffer.alloc(32, acc.privateKey);
```

Comment:
I think you want to use `ethereumjs-utils`' `toBuffer` instead

Reply:
Eek. sorry. `toBuffer` was the fix. Will re-open the PR and try to push to it.